numpyro/infer/util.py

```diff
@@ -217,8 +217,6 @@ def _unconstrain_reparam(params, site):
         log_det = sum_rightmost(
             log_det, jnp.ndim(log_det) - jnp.ndim(value) + len(site["fn"].event_shape)
         )
-        if site["scale"] is not None:
-            log_det = site["scale"] * log_det
```

Comment:
@fehiepsi can you please explain how exactly this logic effects our autoguide?

Reply:
i'd like to understand better where exactly this bug would be triggered. only inside a plate? both factor and sample? etc. in particular i'd like to understand which different code path in local.py avoided this issue...

Reply:
in any case the test now behaves as expected : ) thanks! 💯

Reply:
As in the new tests that I added, when we do unconstrained reparam, we introduce new factor terms. Those factor terms will belong to the same context as the corresponding latent variables, i.e. they will have the same `scale` field in their messages. So with those lines, we are scaling logdet terms 2 times.

Reply:
ah gotcha i think that makes sense